service/smithyprototype/lexruntimeservice/api_op_PutSession.go

```diff
@@ -390,3 +395,103 @@ type PutSessionResponse struct {
 func (r *PutSessionResponse) SDKResponseMetdata() *aws.Response {
 	return r.response
 }
+
+type putSessionSerializeMiddleware struct{}
+
+// ID is the middleware identifier
+func (p putSessionSerializeMiddleware) ID() string {
+	return "PutSessionSerializeMiddleware"
+}
+
+func (p putSessionSerializeMiddleware) HandleSerialize(ctx context.Context, in middleware.SerializeInput, next middleware.SerializeHandler) (
+	out middleware.SerializeOutput, metadata middleware.Metadata, err error,
+) {
+	request, ok := in.Request.(*smithyHTTP.Request)
+	if !ok {
+		return out, metadata, fmt.Errorf("unknown transport type %T", in.Request)
+	}
+
+	input, ok := in.Parameters.(*PutSessionInput)
+	if !ok {
+		return out, metadata, fmt.Errorf("unknown input parameters type %T", in.Parameters)
+	}
+
+	restEncoder := rest.NewEncoder(request.Request)
+
+	restEncoder.AddHeader("Content-Type").String("application/json")
+
+	if err := serializePutSessionInputAWSREST(input, restEncoder); err != nil {
+		return out, metadata, err
+	}
+
+	jsonEncoder := json.NewEncoder()
+	if err := serializePutSessionInputAWSJSON(input, jsonEncoder.Value); err != nil {
+		return out, metadata, err
+	}
+
+	request.Stream = bytes.NewReader(jsonEncoder.Bytes())
+
+	return next.HandleSerialize(ctx, in)
+}
+
+func serializePutSessionInputAWSJSON(v *PutSessionInput, encoder json.Value) error {
+	object := encoder.Object()
+	defer object.Close()
+
+	if v == nil {
+		return nil
+	}
+
```

Comment:
I believe this was commented before, but wanted to reiterate. 

if v == nil is even possible, there shouldn't be a serializing function generated for that shape.

Reply:
Will add error to these ones, the previous one was around whether encoding actual json null vs error for other types.